### ISSUE TITLE
Retrieve the latest ami_release_version for managed node groups

### ISSUE THREAD

**Issue by cebidhem** [terraform-aws-modules/terraform-aws-eks#2589]
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - No 🛑: please wait to file a request until the functionality is avaialble in the AWS provider
  - Yes ✅: please list the AWS provider version which introduced this functionality

Not really.

## Is your request related to a problem? Please describe.
As described in https://github.com/terraform-aws-modules/terraform-aws-eks/issues/2434 , it appears that we can't really managed the version for the AMIs in EKS managed node group. At least, we can't update them from terraform.

While there might be a workaround by using the `ami_id`, this would be considered as custom template.

## Describe the solution you'd like.
Following the closed issue above led me to an other issue opened by @bryantbiggs here -> https://github.com/aws/containers-roadmap/issues/1945.

One of the comment proposed to work around this issue by getting the release_version as a data from SSM: https://github.com/aws/containers-roadmap/issues/1945#issuecomment-1464681858 

Which seems to give the expected value.

## Describe alternatives you've considered.
Update the managed node group through the AWS Console.

**Comment by kaykhan** [comment]
Hi @cebidhem, what did you end up doing. We have been asked by aws support to upgrade our AMI release version due to a bug which has been resoleved in a later release. 

> After the investigation, internal team mentioned that the worker node AMI's kernel (version: 5.10.176-157.645.amzn2) is experiencing some issues. In the most recent AMI version, they have updated the kernel version and added the fix for these issues.

Just curious what is the path for us to be able to do this (what did you end up doing?).  Our current managed node group looks like the following.

```
   "${local.cluster_name}-cron" = {
      iam_role_use_name_prefix = false
      ami_type                 = "AL2_x86_64"
      min_size                 = 1
      max_size                 = 5
      desired_size             = 1
      instance_types           = ["m5.xlarge"]
      capacity_type            = "ON_DEMAND"
      ebs_optimized            = true
      labels = {
        "geeiq/node-type" = "cron"
        Environment       = local.workspace
      }

      block_device_mappings = {
        xvda = {
          device_name = "/dev/xvda"
          ebs = {
            volume_size           = 300
            volume_type           = "gp3"
            delete_on_termination = true
          }
        }
      }

      tags = local.tags
    },
  }
```

**Comment by cebidhem** [comment]
Hi @kaykhan 👋 

Basically I run a script that calls `aws ssm get-parameters-by-path --path /aws/service/bottlerocket/aws-k8s-1.22/x86_64/latest/ --region us-east-1 --recursive` (with the right values) and I input in the code this way:
```terraform
eks_managed_node_group_defaults = {
    tags                = local.mng_tags
    desired_size        = 0
    min_size            = 0
    max_size            = 100
    capacity_type       = "ON_DEMAND"
    ami_type            = "BOTTLEROCKET_x86_64"
    platform            = "bottlerocket"
    ami_release_version = "1.13.5-33225cc9"
}
```
I'm using it in the defaults, but you could add a line `ami_release_version = "1.13.5-33225cc9"` in the nodegroup config you've pasted in your comment.

Honestly, far from being ideal but coming up with a cleaner way has remained in my backlog so far, and I don't update them as often as I probably should.

**Comment by kaykhan** [comment]
> 

Ah okay thanks, just a follow up. I read on the other issue that it required recreating the launch template/asg. Did you experience that? This seems like just an update to the existing one

**Comment by scotlyt** [comment]
Following up on this, I believe the issues were because the release_version on the terraform resource aws_eks_node_group isn't controlled by the module or at least not specifically called out in the examples.

The example in this repo at this current time does not specifically call out that you need to set the ami_release_version

https://github.com/terraform-aws-modules/terraform-aws-eks/blob/v19.21.0/examples/complete/main.tf#L158

```
eks_managed_node_group_defaults = {
    ami_type       = "AL2_x86_64"
    instance_types = ["m6i.large", "m5.large", "m5n.large", "m5zn.large"]

    attach_cluster_primary_security_group = true
    vpc_security_group_ids                = [aws_security_group.additional.id]
    iam_role_additional_policies = {
      additional = aws_iam_policy.additional.arn
    }
}
```

I pass in variables into this module from my own module (better version control across my dev/prod environments)

```
eks_managed_node_group_defaults = {
    ami_type = var.eks_ami_type
    ami_release_version = var.eks_ami_release_version
    instance_types = ["m5.large"]
}
```

The variable I get is dependent on the ami_type & current release version referenced here: https://github.com/awslabs/amazon-eks-ami/blob/master/CHANGELOG.md

This allowed me to pass in my own terraform.tfvars from my module into terraform-aws-eks

```
eks_ami_type = "AL2_x86_64"
eks_ami_release_version = "1.28.5-20240110"
```

and the node groups was able to handle the update the AMI update process without recreating the launch templates or saying that it couldn't handle the work. If you try and change the ami_type and pass in the AMI that amazon publishes, it gets treated as a custom ami, so there needs some documentation/naming to fix this so people can continue to use the published AMIs

**Comment by antonbabenko** [comment]
This issue has been resolved in version 20.6.0 :tada:
